awscli/customizations/datapipeline/__init__.py

```diff
@@ -400,9 +399,18 @@ def _list_runs(self, parsed_args, parsed_globals):
         formatter(self.NAME, converted)
 
     def _describe_objects(self, pipeline_id, object_ids):
-        parsed = self.client.describe_objects(
-            pipelineId=pipeline_id, objectIds=object_ids)
-        return parsed
+        # DescribeObjects will only accept 100 objectIds at a time,
+        # so we need to break up the list passed in into chunks that are at
+        # most that size. We then aggregate the results to return.
+        objects = []
+        max_items_per_describe = 100
+        for i in range(0, len(object_ids), max_items_per_describe):
```

Comment:
I think this is still going to lop items off the end isn't it? If there are 202 items, then this range would return: ``[0, 100, 200]``, missing the last two?

Reply:
That would result in three requests: `0-99, 100-199, 200-202`

Reply:
The tests should cover this since i'm testing a set of size 150.

Reply:
Ah yes, you're correct. I had the logic switched around in my head, thinking the range was returning the upper limit not the lower limit of the object IDs.